frontend/src/metabase/nav/containers/CollectionBreadcrumbs/CollectionBreadcrumbs.tsx

```diff
@@ -1,19 +1,72 @@
-import Collections from "metabase/entities/collections";
-import { getCollectionId } from "metabase/selectors/app";
-import type { State } from "metabase-types/store";
-
-import CollectionBreadcrumbs from "../../components/CollectionBreadcrumbs";
-
-const collectionProps = {
-  id: (state: State, props: { collectionId?: string }) => {
-    if (props.collectionId) {
-      return props.collectionId;
-    }
-    return getCollectionId(state) ?? "root";
+import { type WithRouterProps, withRouter } from "react-router";
+import _ from "underscore";
+
+import {
+  skipToken,
+  useGetCardQuery,
+  useGetCollectionQuery,
+  useGetDashboardQuery,
+} from "metabase/api";
+import { ROOT_COLLECTION } from "metabase/entities/collections";
+import { useSelector } from "metabase/lib/redux";
+import { getViewedItem } from "metabase/selectors/app";
+import type { CollectionId } from "metabase-types/api";
+
+import {
+  CollectionBreadcrumbs as Breadcrumbs,
+  type CollectionBreadcrumbsProps as BreadcrumbsProps,
+} from "../../components/CollectionBreadcrumbs/CollectionBreadcrumbs";
+
+type CollectionBreadcrumbsProps = WithRouterProps &
+  Omit<BreadcrumbsProps, "collection" | "dashboard" | "baseCollectionId"> & {
+    collectionId?: CollectionId;
+    baseCollectionId?: CollectionId | null;
+  };
+
+// TODO: not sure how these changes will work for the embedding folks
```

Comment:
Note to self to fix: I need to try reverting this to a stateful version I had previously committed then reverted.